### PR TITLE
Provide a back button from the Restore process to the lockscreen

### PR DESCRIPTION
**Pull request by mankyKitty**

### CODE REVIEW

File: desktop/src/Desktop/Setup.hs [hunk 1]
```diff
@@ -187,9 +187,20 @@ walletSetupRecoverHeader currentScreen = setupDiv "workflow-header" $ do
...
+    , MonadFix m
+    , MonadHold t m
+    , MonadIO m
+    , PerformEvent t m
+    , PostBuild t m
+    , MonadJSM (Performable m)
+    , TriggerEvent t m
+    )
+  => Bool
+  -- -> m (Event t (Crypto.XPrv, Password))
```
Comment (alexfmpe) [inline desktop/src/Desktop/Setup.hs]: Leftover

File: desktop/src/Desktop/Setup.hs [hunk 2]
```diff
@@ -202,11 +213,18 @@ runSetup = setupDiv "fullscreen" $ mdo
...
+  pure $ leftmost
+    [ fmap Right $ switchDyn $ snd <$> dwf
+    , Left <$> eBack
+    ]
   where
-    hideBack ws
-      | ws `elem` [WalletScreen_SplashScreen, WalletScreen_Done] = setupClass "hide"
-      | otherwise = setupScreenClass ws
+    hideBack ws =
+      if showBackOverride then
```
Comment (alexfmpe) [inline desktop/src/Desktop/Setup.hs]:
```haskell
if not showBackOverride && (ws `elem` [WalletScreen_SplashScreen, WalletScreen_Done]) then
  setupClass "hide"
else
  setupScreenClass ws
```

File: desktop/src/Desktop/Frontend.hs [hunk 3]
```diff
@@ -123,18 +128,28 @@ bipWallet
...
 bipWallet appCfg = do
-  mRoot <- getItemStorage localStorage BIPStorage_RootKey
-  rec
-    root <- holdDyn (mRoot, Nothing) upd
-    upd <- switchHold never <=< dyn $ ffor root $ \case
-      (Nothing, _) -> do
-        keyAndPass <- runSetup
-        performEvent $ ffor keyAndPass $ \(x, Password p) -> do
+  let
+    runSetup0 r fromLockScreen = do
```
Comment (alexfmpe) [inline desktop/src/Desktop/Frontend.hs:132]: 0 ?

Comment (mankyKitty) [reply desktop/src/Desktop/Frontend.hs:132]: It's either that or `runSetup'`, I just needed a name.

**Pull request merged**